### PR TITLE
Use sigs.k8s.io/yaml instead of gopkg.in/yaml

### PR DESCRIPTION
**Pull request by skitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Since we have a Kubernetes-specific fork of go-yaml, use that consistently throughout the project. This doesn't eliminate the dependencies on gopkg.in/yaml, which are still used indirectly; but it ensures that the whole project benefits from fixes or changes to sigs.k8s.io/yaml.

Instead of updating the reference in applyconfiguration-gen (`yamlMarshal`), drop it since it’s unused.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by brianpursley** [comment]
Someone correct me if I'm wrong, but I thought we wanted to move away from maintaining a custom yaml library.

I seem to recall some issues with yaml separators or something like that maybe with Kustomize, but I can't remember xactly what the issue was off the top of my head.

**Comment by skitt** [comment]
> Someone correct me if I'm wrong, but I thought we wanted to move away from maintaining a custom yaml library.
> 
> I seem to recall some issues with yaml separators or something like that maybe with Kustomize, but I can't remember xactly what the issue was off the top of my head.

Thanks for the feedback — this is part of #127888 and based on https://github.com/kubernetes-sigs/yaml/pull/76, but perhaps things have changed since then.

**Comment by brianpursley** [comment]
Ok, I could be mistaken then.  Just want to make sure this change had a purpose. Thanks for the context.

**Comment by BenTheElder** [comment]
If anything we've wound up going deeper on sigs.k8s.io/yaml having forked yaml library code.

Posting this to the code-organization channel for more opinions.

File: staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/types.go [hunk 1]
```diff
@@ -33,5 +33,4 @@ var (
 	smdParser            = types.Ref("sigs.k8s.io/structured-merge-diff/v4/typed", "Parser")
 	testingTypeConverter = types.Ref("k8s.io/client-go/testing", "TypeConverter")
 	yamlObject           = types.Ref("sigs.k8s.io/structured-merge-diff/v4/typed", "YAMLObject")
-	yamlUnmarshal        = types.Ref("gopkg.in/yaml.v2", "Unmarshal")
```
[the remark this answers is not in the document]
Comment (skitt) [reply staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/types.go:36]: Yes; I mentioned it in [the commit message](https://github.com/kubernetes/kubernetes/pull/127901/commits/aca3214a701a2164861fca075b791feaf597b45a) but forgot to update the PR description.

**Comment by aojea** [comment]
This https://github.com/kubernetes/kube-openapi/pull/511 and https://github.com/kubernetes-sigs/structured-merge-diff/pull/264 will also help to remove some of the indirect dependencies

Flake is https://github.com/kubernetes/kubernetes/issues/127858

/test pull-kubernetes-integration

/lgtm

File: staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dims) [reply staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/types.go:36]: thanks!

**Comment by aojea** [comment]
> {Failed;  === RUN   TestPeerProxiedRequestToThirdServerAfterFirstDies

/test pull-kubernetes-integration

**Comment by neolit123** [comment]
/retest
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127901/pull-kubernetes-integration/1843775697982590976
- https://github.com/kubernetes/kubernetes/issues/127858

**Comment by aojea** [comment]
/assign @thockin 

@skitt FYI https://github.com/kubernetes/kube-openapi/pull/511 and https://github.com/kubernetes-sigs/structured-merge-diff/pull/264

**Comment by skitt** [comment]
> @skitt FYI [kubernetes/kube-openapi#511](https://github.com/kubernetes/kube-openapi/pull/511) and [kubernetes-sigs/structured-merge-diff#264](https://github.com/kubernetes-sigs/structured-merge-diff/pull/264)

I saw those, thanks! We’ll get there, one piece at a time :wink:

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**